File: frontend/src/types/TutorSearchRequest.ts
export interface TutorSearchRequest {
  name?: string;
  subject?: string;
  minPrice?: number;
  maxPrice?: number;
  availability?: string; // e.g., MONDAY, WEEKDAY, EVENING
}

export type DayAvailability = {
  [day: string]: { start: string; end: string; enabled: boolean };
};

export interface Tutor {
  _id: string;
  firstname: string;
  lastname: string;
  subject: string;
  hourlyRate: number;
  availability: DayAvailability;
}

File: backend/src/main/java/com/csy/springbootauthbe/student/utils/TutorSearchRequest.java
package com.csy.springbootauthbe.student.utils;

import lombok.Data;

@Data
public class TutorSearchRequest {
    private String name;        // partial match (firstname + lastname)
    private String subject;     // exact match
    private Double minPrice;
    private Double maxPrice;
    private String availability; // e.g. "MONDAY" key in availability map
}

